### PR TITLE
Rework Device Plugin To Handle Multiple Device Types

### PR DESCRIPTION
**Pull request by wizhaoredhat**

### CODE REVIEW

File: daemon/device-plugin/deviceplugin.go [hunk 1]
```diff
@@ -311,10 +265,11 @@ func (nf *nfResources) GetDevicePluginOptions(ctx context.Context, empty *plugin
...
-func NewGrpcPlugin() *nfResources {
+func NewDevicePlugin(dh DeviceHandler) *nfResources {
 	return &nfResources{
-		log:        ctrl.Log.WithName("DevicePlugin"),
-		devices:    make(map[string]pluginapi.Device),
-		socketFile: pluginEndpoint,
+		log:           ctrl.Log.WithName("DevicePlugin"),
+		devices:       make(map[string]pluginapi.Device),
+		socketFile:    pluginEndpoint,
+		deviceHandler: dh,
```
Comment (bn222) [inline daemon/device-plugin/deviceplugin.go:270]: don't abbreviate

File: daemon/device-plugin/deviceplugin.go [hunk 2]
```diff
@@ -311,10 +265,11 @@ func (nf *nfResources) GetDevicePluginOptions(ctx context.Context, empty *plugin
 	}, nil
 }
 
-func NewGrpcPlugin() *nfResources {
+func NewDevicePlugin(dh DeviceHandler) *nfResources {
```
Comment (bn222) [inline daemon/device-plugin/deviceplugin.go]: Would it make sense to not get deviceHandler as an argument?

Comment (wizhaoredhat) [reply daemon/device-plugin/deviceplugin.go]: Sorry what do you mean? We want to pass the device handler into the device plugin.

Comment (wizhaoredhat) [reply daemon/device-plugin/deviceplugin.go]: We agreed to fix this along with main_test.go fixes.

**Pull request merged**